src/gui/src/HydrogenApp.cpp

```diff
@@ -636,34 +639,57 @@ void HydrogenApp::onEventQueueTimer()
 	}
 
 	// midi notes
-	while(!pQueue->m_addMidiNoteVector.empty()){
-
-		int rounds = 1;
-		if(pQueue->m_addMidiNoteVector[0].b_noteExist) { // run twice, delete old note and add new note. this let the undo stack consistent 
-			rounds = 2;
+	while( !pQueue->m_addMidiNoteVector.empty() ){
+		Song *pSong = Hydrogen::get_instance()->getSong();
+		Instrument *pSelectedInstrument = pSong->getInstrumentList()->get( pQueue->m_addMidiNoteVector[0].m_row );
+		// find if a (pitch matching) note is already present
+		Note *pOldNote = pSong->getPatternList()->get( Hydrogen::get_instance()->getSelectedPatternNumber() )
```

Comment:
Do you use the current pattern over the one stored in `pQueue->m_addMidiNoteVector[0].m_pattern` on purpose?

Reply:
Yes